main.js

```diff
@@ -56,40 +58,51 @@ startCompleteButton.addEventListener("click", callStartTimer);
 
 logActivityButton.addEventListener("click", logActivity);
 
+createNewActivityButton.addEventListener("click", returnHome);
+
+window.addEventListener("load", createActivityCard);
+
 // FUNCTIONS
 
+function returnHome(){
+  window.location.reload();
+}
+
 function logActivity() {
   addHidden(timerViewPage);
   addHidden(noActivitiesText);
   removeHidden(completedViewPage);
-  createActivityCard(currentActivity);
   pageTitle.innerText = "Completed Activity";
-  // goal: display the saved currentActivity on a card in the correct area
-  // input: referencing currentActivity object
-  // output: a new activity card with data from currentActivity object
-  // steps: add hidden to timer view and 'you havent logged' Text
-  //        remove hidden from completed view
-  //        display currentActivity data on card (helper function creatActivityCard???)
+  currentActivity.saveToStorage();
+  createActivityCard();
 };
 
-function createActivityCard(currentActivity) {
 
+function createActivityCard() {
   activityCardSection.innerHTML = ``;
-  activityCardSection.innerHTML = `
-  <div class="activity-card">
-    <div class="activity-details">
-      <p class="activity-card-label">${currentActivity.category}</p>
-      <p class="activity-card-time">${currentActivity.minutes.toString().padStart(2, "0")}:${currentActivity.seconds.toString().padStart(2, "0")}</p>
-      <p class="activity-card-description">${currentActivity.description}</p>
-    </div>
-    <div class="activity-icon-div">
-      <div class="activity-icon" id ="${currentActivity.id}">
+  for (var i = 0; i < localStorage.length; i++)
+  if (localStorage.key(i).includes("activity")) {
```

Comment:
Is there any instance in which the localStorage key will NOT contain the string `activity` ? I'm not seeing one from my perusal of the code. I think in this instance you probably only need the for loop and you could remove this if statement.

In regards to the for loop - the way you've written it here is certainly functional, but I do think it is good practice to wrap the code effected by the for loop in curly braces. 

Technically if the for loop only has a single statement you can write it without curly braces (in this case the `if statement` is acting as a single statement) but it is good practice to use the curly brace to better mark out the scope of the for loop and it is good practice in terms of developer empathy - it just makes it easier to read! 

If you remove the if statement here, you will need curly braces around 85-102 to avoid errors in the console when local storage is empty.

Reply:
Since we are storing each activity individually in localStorage and looping through all keys in localStorage - if a user has anything else in local storage, the page will throw an error message and the saved activities will not load. So I think the for loop is necessary to avoid it in our case. I think future edit would be trying to store the activities in LS inside one array, rather than all individually - that would also avoid an error if there's anything else in LS without having to check with the for loop.